common/src/main/scala/uk/ac/wellcome/sns/SNSWriter.scala

```diff
@@ -6,32 +6,55 @@ import com.google.inject.Inject
 import com.twitter.inject.Logging
 import uk.ac.wellcome.models.aws.SNSConfig
 import uk.ac.wellcome.utils.GlobalExecutionContext.context
+import uk.ac.wellcome.models.aws.S3Config
+import com.amazonaws.services.s3.AmazonS3
+import java.util.UUID.randomUUID
 
 import scala.concurrent.{blocking, Future}
 
-case class PublishAttempt(id: Either[Throwable, String])
-
-class SNSWriter @Inject()(snsClient: AmazonSNS, snsConfig: SNSConfig)
-    extends Logging {
-
-  def writeMessage(message: String, subject: String): Future[PublishAttempt] =
-    Future {
-      blocking {
-        debug(
-          s"about to publish message $message on the SNS topic ${snsConfig.topicArn}")
-        snsClient.publish(
-          toPublishRequest(message = message, subject = subject))
-      }
-    }.map { publishResult =>
-        info(s"Published message ${publishResult.getMessageId}")
-        PublishAttempt(Right(publishResult.getMessageId))
-      }
-      .recover {
-        case e: Throwable =>
-          error("Failed to publish message", e)
-          throw e
-      }
-
-  private def toPublishRequest(message: String, subject: String) =
-    new PublishRequest(snsConfig.topicArn, message, subject)
+case class PublishAttempt(id: String, key: String)
+
+class SNSWriter @Inject()(
+  client: AmazonSNS,
+  config: SNSConfig,
+  s3Client: AmazonS3,
+  s3Config: S3Config
```

Comment:
I think the design that we came up with is not really to modify `SNSWriter` to write to s3 but rather to write a wrapper object (we haven't really decided on a name yet, something like `SNSPointerWriter`) that uses `SNSWriter` to send to a SNS message and `S3ObjectStore` to write to s3. So `SNSWriter` probably shouldn't change much

Reply:
Paging @alexwlchan and @kenoir just to make sure we're all on the same page

Reply:
Yeah, this looks wrong to me. I was thinking the existing SNSWriter would stay the same, and we'd add an entirely new class.